src/Rtc/Create.tsx

```diff
@@ -124,7 +128,53 @@ const Create = ({
     return () => {
       engine.current!.destroy();
     };
-  }, []);
+  }, [rtcProps.appId]);
+
+  useEffect(() => {
+    const toggleRole = async () => {
+      if (rtcProps.mode === mode.Live) {
+        if (rtcProps.role == role.Host) {
+          await engine.current?.setClientRole(ClientRole.Broadcaster);
+          // This creates local audio and video track
+          await engine.current?.enableVideo();
+          // This unpublishes the current track
+          await engine.current?.muteLocalAudioStream(true);
+          await engine.current?.muteLocalVideoStream(true);
+          // This updates the uid interface
+          dispatch({
+            type: 'LocalMuteAudio',
+            value: [ToggleState.disabled],
+          });
+          dispatch({
+            type: 'LocalMuteVideo',
+            value: [ToggleState.disabled],
+          });
+        }
+        if (rtcProps.role === role.Audience) {
+          // To switch the user role back to "audience", call unpublish first
+          // Otherwise the setClientRole method call fails and throws an exception.
+          await engine.current?.muteLocalAudioStream(true);
+          await engine.current?.muteLocalVideoStream(true);
+          dispatch({
+            type: 'LocalMuteAudio',
+            value: [ToggleState.disabled],
+          });
+          dispatch({
+            type: 'LocalMuteVideo',
+            value: [ToggleState.disabled],
+          });
+          await engine.current?.setClientRole(ClientRole.Audience);
+        }
+      }
+    };
+    // The first update current skips the render of this block for the first time
+    if (firstUpdate.current) {
+      firstUpdate.current = false;
+      return;
+    }
+    toggleRole();
+  }, [rtcProps.role]);
```

Comment:
role is reactive, but mode isn't. This is a bug. You will get a stale value in your `if` conditions

Reply:
I have removed mode from props as is no longer in state/rtc props